scripts/verify.py

```diff
@@ -36,10 +36,13 @@ def validate(segment_file, sonde_info):
 
         warnings = list(checker.check_segment(seg, seg_navdata, sondes_by_flag))
         for warning in warnings:
-            segmentlogger.warn(warning)
+            if "segment_id" in seg:
+                segmentlogger.warning(seg["segment_id"])
+            segmentlogger.warning(warning)
 
         segment_warning_count += len(warnings)
 
+    navdata.close()
```

Comment:
This is a very good addition. However, I'd rather change that to be using the [`closing` context manager](https://docs.python.org/3/library/contextlib.html#contextlib.closing) e.g.:
```python
from contextlib import closing
...
with closing(get_navdata(...).load()) as navdata:
    ....
```

Reply:
Excellent idea, I did this.